cub/test/test_nvtx_standalone.cu

```diff
@@ -0,0 +1,27 @@
+// The purpose of this test is to verify that CUB can use NVTX without any additional dependencies. It is built as part
+// of the unit tests, but can also be built standalone:
+
+// Compile (from current directory):
```

Comment:
Question: did you verify that this builds in CI with the right command line options?

Reply:
Right. So I considered hooking up this test as a special minimal build analogous to the header checks like we discussed offline. However, I noticed the header checks also link against the cub target in the same way as the non-Catch2 unit tests do. So adding this standalone test as a non-Catch2 test should be equally minimal.

For reference here are the compile and link command lines for building `VERBOSE=1 make cub.cpp17.test.nvtx_standalone`. Compile:
```
/usr/local/cuda/bin/nvcc
-forward-unknown-to-host-compiler
-ccbin=clang++-17
-DCUB_DETAIL_DEBUG_ENABLE_SYNC
-DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CUDA
-DTHRUST_HOST_SYSTEM=THRUST_HOST_SYSTEM_CPP
-D_CCCL_NO_SYSTEM_HEADER
-I/home/bgruber/dev/cccl/cub/test
-I/home/bgruber/dev/cccl/cub/cub/cmake/../..
-I/home/bgruber/dev/cccl/libcudacxx/lib/cmake/libcudacxx/../../../include
-I/home/bgruber/dev/cccl/thrust/thrust/cmake/../..
-O2
-g
-DNDEBUG
-std=c++17
"--generate-code=arch=compute_89,code=[compute_89,sm_89]"
-Xcompiler=-Werror
-Xcompiler=-Wall
-Xcompiler=-Wextra
-Xcompiler=-Winit-self
-Xcompiler=-Woverloaded-virtual
-Xcompiler=-Wcast-qual
-Xcompiler=-Wpointer-arith
-Xcompiler=-Wunused-local-typedef
-Xcompiler=-Wvla
-Xcompiler=-Wgnu
-Xcompiler=-Wno-gnu-line-marker
-Xcompiler=-Wno-gnu-zero-variadic-macro-arguments
-Xcompiler=-Wno-unused-function
-Wreorder
-Xcudafe=--display_error_number
-Xcudafe=--promote_warnings
-Wno-deprecated-gpu-targets
-MD
-MT
cub/test/CMakeFiles/cub.cpp17.test.nvtx_standalone.dir/test_nvtx_standalone.cu.o
-MF
CMakeFiles/cub.cpp17.test.nvtx_standalone.dir/test_nvtx_standalone.cu.o.d
-x
cu
-c
/home/bgruber/dev/cccl/cub/test/test_nvtx_standalone.cu
-o
CMakeFiles/cub.cpp17.test.nvtx_standalone.dir/test_nvtx_standalone.cu.o
```
Link:
```
/usr/bin/clang++-17
CMakeFiles/cub.cpp17.test.nvtx_standalone.dir/test_nvtx_standalone.cu.o
-o
../../bin/cub.cpp17.test.nvtx_standalone
-lcudadevrt
-lcudart_static
-lrt
-lpthread
-ldl
-L"/usr/local/cuda/targets/x86_64-linux/lib/stubs"
-L"/usr/local/cuda/targets/x86_64-linux/lib"
```

However, the test compiles and runs equally well if I just compile it with:
`nvcc test_nvtx_standalone.cu -I../../cub -I../../thrust -I../../libcudacxx/include -o nvtx_standalone`